### PR TITLE
disable dragging of bottom sheet in offline mode

### LINKED ISSUE #2679: Disable dragging up in offline mode
image

### PR DESCRIPTION
**Pull request by vanshikaarora**
### Description
Fixes #2679 Disable dragging up in offline mode

**Changes made**
Disabled bottom sheet behavior on dragging in offline mode

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ContinuousScanActivity.java [hunk 1]
```diff
@@ -825,6 +831,18 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         }
     }
 
+    private boolean isNetworkAvailable() {
+        ConnectivityManager manager =
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ContinuousScanActivity.java]: please see:  isNetworkConnected in Utils.

Comment (vanshikaarora) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ContinuousScanActivity.java]: @teolemon I have made the changes. Kindly review :)

Comment (deniger) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ContinuousScanActivity.java]: @vanshikaarora  please, could you remove the unused import ?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ContinuousScanActivity.java [hunk 2]
```diff
@@ -580,6 +582,9 @@ protected void onCreate(Bundle savedInstanceState) {
         bottomSheetBehavior.setBottomSheetCallback(new BottomSheetBehavior.BottomSheetCallback() {
             @Override
             public void onStateChanged(@NonNull View bottomSheet, int newState) {
+                if (!Utils.isNetworkConnected(getBaseContext()) && newState == BottomSheetBehavior.STATE_DRAGGING) {
+                    bottomSheetBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
+                } else {
                 if (newState == BottomSheetBehavior.STATE_HIDDEN) {
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ContinuousScanActivity.java]: Could you indent from here?

**Comment by teolemon** [comment]
@vanshikaarora @Karljoones @deniger small conflict to merge

**Comment by teolemon** [comment]
@philippeauriach this one is related to your latest PR, and intends to disable dragging up

**Comment by teolemon** [comment]
Ok @philippeauriach I'm closing it then.

**Pull request closed without merging**